app/_src/networking/hostnamegenerator.md

```diff
@@ -2,32 +2,139 @@
 title: HostnameGenerator
 ---
 
+{% if_version gte:2.9.x %}
 {% warning %}
 This resource is experimental!
 {% endwarning %}
+{% endif_version %}
 
 A `HostnameGenerator` provides
 
-- a template to generate hostnames from properties of `MeshServices` and `MeshExternalServices`
-- a selector that defines for which `MeshServices` and `MeshExternalServices` this generator runs
+- a template to generate hostnames from properties of `MeshServices`, `MeshMultiZoneService`, and `MeshExternalServices`
+- a selector that defines for which `MeshServices`, `MeshMultiZoneService`, and `MeshExternalServices` this generator runs
 
-{% policy_yaml hg-all %}
+## Defaults
+
+{{site.mesh_product_name}} ships with default HostnameGenerators depending on the control plane mode and storage type.
+
+### Local MeshService in Universal zone
+
+The following policy is automatically created on a zone control plane running in the Universal mode.
+```yaml
+type: HostnameGenerator
+name: local-universal-mesh-service
+spec:
+  selector:
+    meshService:
+      matchLabels:
+        kuma.io/origin: zone
+  template: "{% raw %}{{ .DisplayName }}.svc.mesh.local{% endraw %}"
+```
+
+### Local MeshExternalService
+
+The following policy is automatically created on a zone control plane.
```

Comment:
Just my opinion but maybe each should have an explanation, since they're essentially examples explaining the resource, like "add a hostname `{{ .DisplayName }}.extsvc.mesh.local` for every MeshExternalService in the local zone"

Reply:
how about now?